e2e/test/prerequisites/E2ETestsSetup/e2eTestsArmTemplate.json

```diff
@@ -22,49 +10,72 @@
                 "description": "The name of the storage account used by the IoT hub."
             }
         },
-        "ResourceGroup": {
+        "UserObjectId": {
             "type": "string",
             "metadata": {
-                "description": "The name of the resource group to create all resources used by tests."
+                "description": "Signed in users's objectId"
             }
         },
-        "UserObjectId": {
+        "FarRegion": {
             "type": "string",
+            "defaultValue": "southeastasia",
             "metadata": {
-                "description": "Signed in users's objectId"
+                "description": "The region for the second IoT hub in a DPS that is far away from the test devices."
             }
         },
         "HubName": {
             "type": "string",
+            "defaultValue": "[resourceGroup().id]",
             "metadata": {
                 "description": "The name of the main IoT hub used by tests."
             }
         },
         "FarHubName": {
             "type": "string",
+            "defaultValue": "[concat([=resourceGroup().id, '-far')]",
             "metadata": {
                 "description": "The name of the far away IoT hub used by tests."
             }
         },
         "DeviceProvisioningServiceName": {
             "type": "string",
+            "defaultValue": "[resourceGroup().id]",
             "metadata": {
                 "description": "The name of the DPS used by tests."
             }
+        },
+        "KeyVaultName": {
+            "type": "string",
+            "defaultValue": "concat(resourceGroup().id)",
+            "metadata": {
+                "description": "The name of the key vault for storing secrets needed for running tests."
+            }
+        },
+        "OperationalInsightsName": {
```

Comment:
Moving more things into parameters or variables, so we don't have to find all the cases where we assumed the name would be the rg name.